src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -2145,13 +2154,9 @@ template <typename InterfaceType> void RadioSpinel<InterfaceType>::RestoreProper
                          sizeof(otMacKey)));
     }
 
-    if (mInstance != nullptr)
+    if (mMacFrameCounterSet)
     {
-        if (static_cast<Instance *>(mInstance)->template Get<Settings>().Read(networkInfo) == OT_ERROR_NONE)
-        {
-            SuccessOrDie(
-                Set(SPINEL_PROP_RCP_MAC_FRAME_COUNTER, SPINEL_DATATYPE_UINT32_S, networkInfo.GetMacFrameCounter()));
-        }
+        SuccessOrDie(Set(SPINEL_PROP_RCP_MAC_FRAME_COUNTER, SPINEL_DATATYPE_UINT32_S, mMacFrameCounter));
```

Comment:
There is still some small chance that radio/RCP has used some counter after `mMacFrameCounter` (for enh ack) and they are in queue to be sent to host (not yet processed by host `RadioSpinel`).

So how about having `mMacFrameCounter` track the last used value we see from RCP? And here we add some guard jump when we set it, maybe setting it to `mMacFrameCounter + 10` (jump over 10?).

Reply:
I have updated the code to add the guard.